modules/mapbox/src/mapbox-overlay.ts

```diff
@@ -95,6 +95,14 @@ export default class MapboxOverlay implements IControl {
       gl: map.painter.context.gl
     });
 
+    this._deck = getDeckInstance({
+      map,
+      // @ts-ignore non-public map property
+      gl: map.painter.context.gl,
+      deck: this._deck
+    });
+    this.setProps(this._deck.props);
```

Comment:
Is this still necessary?

Reply:
If the user changes `parameters` at some point later on, they'll be removing the necessary ones set by `getDeckInstance`. How should they access the necessary parameters to keep them intact?

I'm not sure this change addresses the problem - it only synchronizes this._props and this._deck.props, but it'd still be up to the user to spread these internal props into their parameters before calling setProps.

Reply:
I split the prop merging out and call it in `getDeckInstance` and the overlay's `setProps`. I do not attempt to sync `overlay._props` and `overlay._deck.props` anymore, since it shouldn't be necessary because most default props are reapplied whenever a user changes them.